SimG4Core/Application/src/RunManagerMTWorker.cc

```diff
@@ -217,6 +220,12 @@ void RunManagerMTWorker::resetTLS() {
   --n_tls_shutdown_task;
 }
 
+void RunManagerMTWorker::beginRun(edm::EventSetup const& es) {
```

Comment:
I don't believe we can. Some of the initializations are putting items into the thread_local and those would have to be done also in `produce` since we do not know when the OscarMTProducer will see each of the TBB threads. This is one of the reasons for later moving to std::threads controlled directly by OscarMTProducer (this change would be needed to make that possible).